xbmc/addons/binary-addons/AddonInstanceHandler.h

```diff
@@ -22,14 +22,36 @@ namespace ADDON
 class IAddonInstanceHandler
 {
 public:
+  /**
+   * @brief Class constructor for handling add-on instance processes, allowing
+   * an add-on to handle multiple work simultaneously and independently.
+   *
+   * @param[in] type The associated add-on type which is processed in the running
+   *                 instance.
+   * @param[in] addonInfo Class for querying available add-on information (e.g.
+   *                      content declared in addon.xml).
+   * @param[in] parentInstance Above running add-on instance which starts this
+   *                           instance (e.g. inputstream add-on instance opens
+   *                           corresponding and necessary video codec instance).
+   * @param[in] uniqueWorkID Identification value intended to pass any special
+   *                         values to the instance to be opened. Currently used,
+   *                         for example, to transfer special values to inputstream
+   *                         using the property @ref STREAM_PROPERTY_INPUTSTREAM_INSTANCE_ID
+   *                         from external space, for example PVR add-on which also
+   *                         supports inputstream can exchange special values with
+   *                         it, e.g. select the necessary add-on processing class,
+   *                         since it is not known at the start what is being
+   *                         executed ( live TV, radio, recordings...) and add-on
+   *                         may use different classes.
+   */
```

Comment:
@garbear can you look about the text, not sure how it can be better with my bad english :sweat_smile:.

Reply:
Have played a bit (not uploaded now):
```cpp
  /**
   * @brief Class constructor for handling add-on instance processes, allowing
   * an add-on to handle multiple work simultaneously and independently.
   *
   * @param[in] type The associated add-on type which is processed in the running
   *                 instance.
   * @param[in] addonInfo Class for querying available add-on information (e.g.
   *                      content declared in addon.xml).
   * @param[in] parentInstance [opt] Above running add-on instance which starts this
   *                           instance. Used to have the associated class for
   *                           work to open in the add-on.
   *                           Currently used places:
   * | Parent | Target | Description
   * |--------|--------|-------------
   * | @ref kodi::addon::CInstanceInputStream | @ref kodi::addon::CInstanceVideoCodec | In order to be able to access the overlying input stream instance in the video codec created by Kodi on the add-on.
   * @param[in] uniqueWorkID [opt] Identification value intended to pass any special
   *                         values to the instance to be opened.
   *                         If not used, the IAddonInstanceHandler class pointer
   *                         is used as a string.
   *                         Currently used values:
   * | Add-on instance type | Description
   * |----------------------|---------------------
   * | @ref kodi::addon::CInstanceInputStream "Inputstream" | To transfer special values to inputstream using the property @ref STREAM_PROPERTY_INPUTSTREAM_INSTANCE_ID from external space, for example PVR add-on which also supports inputstream can exchange special values with it, e.g. select the necessary add-on processing class, since it is not known at the start what is being executed ( live TV, radio, recordings...) and add-on may use different classes.
   * | All other | The used class pointer of Kodi's @ref IAddonInstanceHandler is used as a value to have an individually different value.
   */
  IAddonInstanceHandler(ADDON_TYPE type,
                        const AddonInfoPtr& addonInfo,
                        KODI_HANDLE parentInstance = nullptr,
                        const std::string& uniqueWorkID = "");
```

Screenshot:
![Screenshot_20220813_111202](https://user-images.githubusercontent.com/6879739/184477311-ec6f3bef-3eb3-419e-9c7a-32c28dfc963a.png)